fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/runners/throttled/ThrottledIteratorTest.java

```diff
@@ -664,10 +686,13 @@ void testIteratorClosesOnNextCloses() throws Exception {
         try (FDBRecordContext context = openContext()) {
             openSimpleRecordStore(context);
             storeBuilder = recordStore.asBuilder();
+            commit(context);
         }
         final ThrottledRetryingIterator.Builder<Integer> builder = ThrottledRetryingIterator.builder(fdb, cursorFactory, itemHandler);
         try (ThrottledRetryingIterator<Integer> iterator = builder.build()) {
             iterateAll = iterator.iterateAll(storeBuilder);
+            // "sleep" for 10 ms to allow futures to get scheduled
+            MoreAsyncUtil.delayedFuture(10, TimeUnit.MILLISECONDS).join();
```

Comment:
Can you solve this with a semaphore instead of sleeping, by having the `cursorFactory` release a semaphore?

Reply:
Done!